lib/TcpConnection.js

```diff
@@ -43,24 +43,29 @@ module.exports = function factory (userOptions) {
 
   // This stream is the one returned to psock.js.
   const outputStream = stream.Writable({
-    close () { socket.end() },
     write (data, encoding, callback) {
       socket.write(data)
       callback()
     }
   })
 
+  outputStream.close = function () {
+    socket.unref()
```

Comment:
why is this `unref()` being added?

Reply:
If ```socket.close()``` is called some of the last messages could not be sent(faced with that issue by myself). ```socket.unref()``` closes all active callbacks and waits until all messages will be sent. After that, the connection is successfully closed.